meerk40t/balormk/device.py

```diff
@@ -1132,16 +1132,20 @@ def balor_goto(command, channel, _, x=None, y=None, remainder=None, **kwgs):
             help=_("Turns redlight on/off"),
         )
         def balor_on(command, channel, _, off=None, remainder=None, **kwgs):
-            if off == "off":
-                reply = self.driver.connection.light_off()
-                self.driver.connection.write_port()
-                self.redlight_preferred = False
-                channel("Turning off redlight.")
-            else:
-                reply = self.driver.connection.light_on()
-                self.driver.connection.write_port()
-                channel("Turning on redlight.")
-                self.redlight_preferred = True
+            try:
```

Comment:
Specific code that directly sends will trigger the ConnectionRefusedError if you can't send that data.